transport/src/main/java/io/netty/channel/nio/AbstractNioChannel.java

```diff
@@ -353,6 +353,7 @@ protected void doRegister() throws Exception {
     @Override
     protected void doDeregister() throws Exception {
         eventLoop().cancel(selectionKey());
+        selectionKey=null;
```

Comment:
Also I wonder if this break some other stuff where we require selectionKey != null with an assert. Maybe it would be better to just do `selectionKey.attach(null)` and so clear the attachment ?